src/components/Button/Button.tsx

```diff
@@ -1,72 +1,60 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import './Button.scss';
 import { ReactComponent as LoadingIndicator } from '../../images/loading.svg';
 
-const propTypes = {
+interface Props {
   /**
-   * A unique identifier for the button
-   */
-  id: PropTypes.string,
+    * A unique identifier for the button
+    */
+  id?: string;
   /**
    * Button type
    */
-  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  type?: 'button' | 'submit' | 'reset';
   /**
-   * Additional ClassNames to add to button
+   * Disables the button, making it inoperable
    */
-  className: PropTypes.string,
+  isDisabled?: boolean;
   /**
-   * Contents of the button
+   * Additional ClassNames to add to button
    */
-  children: PropTypes.node.isRequired,
+  className?: string;
   /**
-   * Disables the button, making it inoperable
+   * Callback when button is pressed
    */
-  isDisabled: PropTypes.bool,
+  fullWidth?: boolean;
   /**
-   * Button takes up the full width of its parent container
+   * Specify the tabIndex of the button
    */
-  isLoading: PropTypes.bool,
+  tabIndex?: number;
   /**
-   * Allow the button to grow to its container's full width
+   * Button takes up the full width of its parent container
    */
-  fullWidth: PropTypes.bool,
+  isLoading?: boolean;
   /**
    * Callback when button is pressed
    */
-  onClick: PropTypes.func,
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
   /**
    * Callback when button receives focus
    */
-  onFocus: PropTypes.func,
+  onFocus?: (e: React.FocusEvent<HTMLButtonElement>) => void;
   /**
    * Callback when focus leaves button
    */
-  onBlur: PropTypes.func,
-  /**
-   * Specify the tabIndex of the button
-   */
-  tabIndex: PropTypes.number,
-};
+  onBlur?: (e: React.FocusEvent<HTMLButtonElement>) => void;
+}
 
-const defaultProps = {
-  id: undefined,
+const defaultProps: Partial<Props> = {
   type: 'button',
   className: '',
   isDisabled: false,
-  isLoading: undefined,
-  fullWidth: undefined,
-  onClick: undefined,
-  tabIndex: undefined,
-  onFocus: undefined,
-  onBlur: undefined,
 };
 
-const Button = ({
+const Button: React.FC<Props> = ({
   id,
-  type,
+  type = 'button',
```

Comment:
We're duplicating the default as `'button`' here, we'll want to get rid of either the param initialization or the default prop! I prefer param initialization since it's just a JS feature and you don't need to do any tinkering to get it to work with TS.

Reply:
youre right! removed the defaultProps object and replaced it with param initialization